src/Nix/Effects/Derivation.hs

```diff
@@ -351,7 +351,7 @@ buildDerivationWithContext drvAttrs = do
     -- common functions, lifted to WithStringContextT
 
     demandF' :: (NValue t f m -> WithStringContextT m a) -> NValue t f m -> WithStringContextT m a
-    demandF' f v = join $ lift $ demandF (pure . f) v
+    demandF' f v = join $ lift $ f <$> demand v
```

Comment:
Note to myself: It seems that demand' -> demandF' was not reverted. For consistency it should probably also be flipped.

Reply:
https://github.com/haskell-nix/hnix/pull/876